lib/bots/versionbot.ts

```diff
@@ -351,6 +350,29 @@ export class VersionBot extends ProcBot {
 				}
 			});
 
+			// We've an event for each PR. However, we now have to check the labels on it and a status event
+			// does not include a PR. As we have already retrieved the right PR here, we filter out any that
+			// have the suppression label on them.
+			return Promise.map(prEvents, (prEvent) => {
+				const pr: GithubApiTypes.PullRequest = prEvent.cookedEvent.data.pull_request;
+				return this.dispatchToEmitter(this.githubEmitterName, {
+					data: {
+						number: pr.number,
+						owner: pr.head.repo.owner.login,
+						repo: pr.head.repo.name,
+					},
+					method: this.githubApi.issues.getIssueLabels
+				}).then((labels: GithubApiTypes.IssueLabel[]) => {
+					// If the suppression labels are present, then we remove the PR from the
+					// list of those to check.
+					if (_.find(labels, (label) => label.name === IgnoreLabel)) {
+						this.logger.log(LogLevel.DEBUG,
+							`Dropping '${registration.name}' as suppression labels are all present`);
+						_.remove(prEvents, (labeledEvent) => labeledEvent === prEvent);
+					}
+				});
+			});
+		}).then(() => {
```

Comment:
If you do the above, you can also actually combine this `then` and the below `Promise.map`, and replace this line with a direct call to map on the existing promise chain:

```typescript
}).map((prEvent) => { ...
```